src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -398,7 +398,7 @@ public SecurityDescriptor getSecurityInfo(SMB2FileId fileId, Set<SecurityInforma
      * The SecurityDescriptor(MS-DTYP 2.4.6 SECURITY_DESCRIPTOR) for the Given FileId
      */
     public void setSecurityInfo(String path, Set<SecurityInformation> securityInfo, SecurityDescriptor securityDescriptor) throws SMBApiException {
-        Set<AccessMask> accessMask = of(GENERIC_WRITE);
+        Set<AccessMask> accessMask = of(READ_CONTROL);
```

Comment:
I would assume we would need to have _some_ write permission for this?

Reply:
The necessary access rights are added just underneath this line. I think we can actually start from an empty set in this method. The only thing that's missing is a check for the 'group owner'. If that's being written I think you also need `WRITE_OWNER`.

Reply:
Isn't that on line 405?

Reply:
Yep, I'll add a review comment there.

Reply:
Yes that's correct. We need to add `|| securityInfo.contains(SecurityInformation. GROUP_SECURITY_INFORMATION)` there I think.